demo/src/main/java/com/novoda/demo/MainActivity.java

```diff
@@ -94,20 +117,27 @@ public void onTracksChanged() {
                 }
             }
         });
+    }
 
+    private void extractFromIntent() {
+        Intent intent = getIntent();
+        if (intent != null) {
+            mpdAddress = Uri.parse(intent.getStringExtra(LandingActivity.KEY_MPD_ADDRESS));
+            licenseServerAddress = intent.getStringExtra(LandingActivity.KEY_LICENSE_SERVER_ADDRESS);
+            downloadLicense = intent.getBooleanExtra(LandingActivity.KEY_DOWNLOAD_LICENSE, false);
+        }
```

Comment:
shall we show a toast indicating that there is an issue? so that the user of the demo knows what needs to be done

Reply:
Toast when any of these are missing? I can do.

Reply:
They'd all cancel each other though if each was missing. I think if it doesn't work it is quite obvious that information is missing from the `LandingActivity`.

Reply:
5626621